sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/DataSourceV2Writer.java

```diff
@@ -30,6 +30,9 @@
  * It can mix in various writing optimization interfaces to speed up the data saving. The actual
  * writing logic is delegated to {@link DataWriter}.
  *
+ * If exception happens when applying any of these writing optimizations, the action would fail and
```

Comment:
an exception.
By the way, not related to this PR, shall we also list allowed writing optimizations like [in DataSourceV2Reader](https://github.com/apache/spark/pull/19623/files#diff-f7f970e9684796e2120223d1153302e4R35) ?

Reply:
we only have one so I don't know how to category them... We can do this after having more optimizations